### ISSUE TITLE
CherryTree 0.99.3 problems on Kubuntu 20.04

### ISSUE THREAD

**Issue by gerdeg** [giuspen/cherrytree#979]
Dear all
I was trying out the experimental version 0.99.3.
-- I can set the background color of the tree, but cannot set the background color of the rich text: my setting has no effect, after restart I get always the same background color.
-- The font size of the right-click menu changes with the font size set in Fonts / Rich text

But more important:
-- The tree hierarchy changes all the time after a change in the order is made and the file is saved and reopened again. That really hurts :-)
 
Operating System: Kubuntu 20.04
KDE Plasma Version: 5.18.5
KDE Frameworks Version: 5.68.0
Qt Version: 5.12.8
Kernel Version: 5.4.0-40-generic
OS Type: 64-bit
Processors: 8 × Intel® Core™ i7-4770 CPU @ 3.40GHz
Memory: 7,7 GiB of RAM

Ger de Gooijer

**Comment by txe** [comment]
About the background, it's a known issue, the approach used in the old cherrytree to change colors doesn't work for Gtk3. The proper fix would take a lot of time, so I skipped it until c++ version testing us finished.

But I didn't get about the tree hierarchy changes, can you elaborate it more?

**Comment by gerdeg** [comment]
What I mean is: main items in the tree switch places 'spontaneously' after saving and reopening, the order of the items changes. The same goes for subitems under a main item. Not something you want in an outliner. I 'll test it out for you and sent you screenshots. Perhaps it has something to do with: these files I used this morning were all produced with the latest version of CherryTree under the latest version of KDE Neon. Perhaps this leads to database corruption or something like that. I will look into it further this weekend.
Ger de Gooijer

**Comment by txe** [comment]
I see, the files must be sqlite, not xml. It's not necessary related to corruptions, it can be some bug with keeping sorting order of nodes. I take a look, but if you know how to reproduce this issue reliably, it would be great.

**Comment by gerdeg** [comment]
Yes, they are ctb-files.

**Comment by txe** [comment]
Although, you wrote it happened every time, so the test maybe unnecessary

**Comment by gerdeg** [comment]
With two files, yes. But, i did not try something like Save and Vacuum and then reopen. And I did not make screenshots to clarify the matter. And I can use other files, I use CherryTree all the time :-)

**Comment by gerdeg** [comment]
So ... I tested four different files and its true:
> If I change the items in the main tree to something else and save and reopen the file the order is different from before the save. That happened with all four test files
> If you reorder the items in a subtree the same happens.
If you want the screenshots for prove I can sent them to you or put them here.

I also discovered: 
> If you switch of the creating of backup copies (Pref/Misc) and restart the function it is switched on again.
> (Sub) Tree items are not wrapped when exceeding the preffered wrapping width of the tree panel.

That's it for today. :-)  Ger de Gooijer

**Comment by gerdeg** [comment]
O, I forgot:
> Save and vacuum does not make a difference in this tree-problem.
> Save As to a different file keeps the hierarchy in the tree the same after an edit. But if you edit this new file the same problem happens and items are spontaneously placed in other positions after saving and reopening..

**Comment by txe** [comment]
Thanks a lot for your help!
I had a lot of work to do the last week, so I was a bit tired from programming. But I think I can start fixing stuff the next week.

**Comment by txe** [comment]
It should be fixed now, thanks for the issues!

**Comment by txe** [comment]
Forgot about font in menu, so it's back

**Comment by txe** [comment]
> The font size of the right-click menu changes with the font size set in Fonts / Rich text

I can't reproduce it, are you sure about this issue?

**Comment by gerdeg** [comment]
Yes, could reproduce it just now. The right-click menu font size outside the CherryTree window is normal, but inside the CherryTree window it takes the font size of Fonts / Right text. Cannot make a screenshot of this, but it is really true here in Kubuntu 20.04.

**Comment by gerdeg** [comment]
Let me know if you have a new version out I can test. Bye now.

**Comment by giuspen** [comment]
0.99.4 is the latest development version packaged but does not include this fix; 0.99.5 will be issued probably next Sunday

**Comment by txe** [comment]
About the issue with the font in the context menu.
It's a bug of the installed system theme and all gtk3 apps are affected by it, e.g. gedit, plume and etc. You can read a bit [here](https://bugs.launchpad.net/ubuntu/+source/pluma/+bug/1823550).

There are three ways to fix it:
- create an issue in kubuntu bugzilla (or that they are using)
- change the current theme
- force the different font for context menu by creating a file `~/.config/gtk-3.0/gtk.css` with
```
.context-menu {
    font-size: 10pt;
    font-family: regular;
}
```
